src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -182,6 +182,32 @@ private bool AllMatches<TSymbolMatcher>(ImmutableArray<TSymbolMatcher> matchers,
             return true;
         }
 
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as SymbolSpecification);
+        }
+
+        public bool Equals(SymbolSpecification other)
+        {
+            if (other is null)
+                return false;
+
+            return ID == other.ID
+                && Name == other.Name
+                && ApplicableSymbolKindList.SequenceEqual(other.ApplicableSymbolKindList)
+                && ApplicableAccessibilityList.SequenceEqual(other.ApplicableAccessibilityList)
+                && RequiredModifierList.SequenceEqual(other.RequiredModifierList);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(ID.GetHashCode(),
+                Hash.Combine(Name.GetHashCode(),
+                    Hash.Combine(Hash.CombineValues(ApplicableSymbolKindList),
+                        Hash.Combine(Hash.CombineValues(ApplicableAccessibilityList),
+                            Hash.CombineValues(RequiredModifierList)))));
+        }
```

Comment:
so why does this help?  clearly we didn't have equals/gethashcode here before... so we coudlnt' have been depending on htat behavior.  so how do this help out?

Reply:
Way up above we serialized the whole tree to XElement and compared those.

Reply:
Ah, yikes.  thanks.